lib/matplotlib/colorbar.py

```diff
@@ -357,34 +426,119 @@ def draw_all(self):
     def config_axis(self):
         ax = self.ax
         if self.orientation == 'vertical':
-            ax.xaxis.set_ticks([])
             # location is either one of 'bottom' or 'top'
             ax.yaxis.set_label_position(self.ticklocation)
             ax.yaxis.set_ticks_position(self.ticklocation)
+            if (isinstance(self.norm, colors.LogNorm)
+                    and self._use_adjustable()):
+                ax.set_xscale('log')
+                ax.set_yscale('log')
+            ax.xaxis.set_ticks([])
+            ax.xaxis.set_ticks([], minor=True)
+
         else:
-            ax.yaxis.set_ticks([])
             # location is either one of 'left' or 'right'
             ax.xaxis.set_label_position(self.ticklocation)
             ax.xaxis.set_ticks_position(self.ticklocation)
+            if (isinstance(self.norm, colors.LogNorm)
+                    and self._use_adjustable()):
+                ax.set_xscale('log')
+                ax.set_yscale('log')
+            ax.yaxis.set_ticks([])
+            ax.yaxis.set_ticks([], minor=True)
 
         self._set_label()
 
+    def _get_ticker_locator_formatter(self):
+        """
+        This code looks at the norm being used by the colorbar
+        and decides what locator and formatter to use.  If ``locator`` has
+        already been set by hand, it just returns
+        ``self.locator, self.formatter``.
+        """
+        locator = self.locator
+        formatter = self.formatter
+        if locator is None:
+            if self.boundaries is None:
+                if isinstance(self.norm, colors.NoNorm):
+                    nv = len(self._values)
+                    base = 1 + int(nv / 10)
+                    locator = ticker.IndexLocator(base=base, offset=0)
+                elif isinstance(self.norm, colors.BoundaryNorm):
+                    b = self.norm.boundaries
+                    locator = ticker.FixedLocator(b, nbins=10)
+                elif isinstance(self.norm, colors.LogNorm):
+                    locator = ColorbarLogLocator(self)
+                elif isinstance(self.norm, colors.SymLogNorm):
+                    # The subs setting here should be replaced
+                    # by logic in the locator.
+                    locator = ticker.SymmetricalLogLocator(
+                                      subs=np.arange(1, 10),
+                                      linthresh=self.norm.linthresh,
+                                      base=10)
+                else:
+                    if mpl.rcParams['_internal.classic_mode']:
+                        locator = ticker.MaxNLocator()
+                    else:
+                        locator = ColorbarAutoLocator(self)
+            else:
+                b = self._boundaries[self._inside]
+                locator = ticker.FixedLocator(b, nbins=10)
+        _log.debug('locator: %r', locator)
+        return locator, formatter
+
+    def _use_adjustable(self):
```

Comment:
I don't have a good suggestion yet (or maybe ever), but just as a thought: maybe there is a better name than 'adjustable' for the alternative style of handling the locator that you are testing for here.

Reply:
Changed to `_use_auto_colorbar_locator`